iceoryx_posh/test/moduletests/test_popo_unique_port_id.cpp

```diff
@@ -26,160 +28,155 @@ using namespace ::testing;
 using namespace iox::popo;
 using namespace iox::cxx;
 
-TEST(TypedUniqueId_RouDiId, SettingTheRouDiIdWorks)
+TEST(UniquePortId_test, SettingTheRouDiIdWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "473467bf-1a6f-4cd2-acd8-447a623a5301");
-    uint16_t someId = 1243u;
-    iox::popo::internal::setUniqueRouDiId(someId);
-    EXPECT_EQ(iox::popo::internal::getUniqueRouDiId(), someId);
-    iox::popo::internal::unsetUniqueRouDiId();
+    uint16_t someId = 1243U;
+    // we cannot ensure that setUniqueRouDiId wasn't called before, therefore we ignore the error
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler([](auto, auto, auto) {});
+    iox::popo::UniquePortId::setUniqueRouDiId(someId);
+    EXPECT_EQ(iox::popo::UniquePortId::getUniqueRouDiId(), someId);
+
+    // reset unique RouDi ID
+    iox::popo::UniquePortId::setUniqueRouDiId(iox::roudi::DEFAULT_UNIQUE_ROUDI_ID);
 }
 
-TEST(TypedUniqueId_RouDiId, SettingTheRouDiIdTwiceFails)
+TEST(UniquePortId_test, SettingTheRouDiIdTwiceFails)
 {
     ::testing::Test::RecordProperty("TEST_ID", "fe468314-cd38-4363-bbf9-f106bf9ec1f4");
-    uint16_t someId = 1243u;
-    bool errorHandlerCalled = false;
+    uint16_t someId = 1243U;
+    optional<iox::Error> detectedError;
+    optional<iox::ErrorLevel> detectedErrorLevel;
     auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
-        [&errorHandlerCalled](const iox::Error error IOX_MAYBE_UNUSED,
-                              const std::function<void()>,
-                              const iox::ErrorLevel) { errorHandlerCalled = true; });
-
-    iox::popo::internal::setUniqueRouDiId(someId);
-    EXPECT_FALSE(errorHandlerCalled);
-    iox::popo::internal::setUniqueRouDiId(someId);
-    EXPECT_TRUE(errorHandlerCalled);
-    iox::popo::internal::unsetUniqueRouDiId();
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            detectedError.emplace(error);
+            detectedErrorLevel.emplace(errorLevel);
+        });
+
+    iox::popo::UniquePortId::setUniqueRouDiId(someId);
+    // we don't know if setUniqueRouDiId was called before, therefore ignore any error
+    detectedError.reset();
+    detectedErrorLevel.reset();
+
+    iox::popo::UniquePortId::setUniqueRouDiId(someId);
+    // now we know that setUniqueRouDiId was called and therefore the error handler must also be called
+    ASSERT_TRUE(detectedError.has_value());
+    ASSERT_TRUE(detectedErrorLevel.has_value());
+    EXPECT_THAT(detectedError.value(),
+                Eq(iox::Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_CUSTOM_UNIQUE_ID));
+    EXPECT_THAT(detectedErrorLevel.value(), Eq(iox::ErrorLevel::SEVERE));
+
+    // reset unique RouDi ID
+    iox::popo::UniquePortId::setUniqueRouDiId(iox::roudi::DEFAULT_UNIQUE_ROUDI_ID);
```

Comment:
Could you turn those `TEST` into test fixtures `TEST_F` and add this line to `void TearDown()`. Then it does not get forgotten and as it seems it is part of the cleanup in every test.

Reply:
Well, I count only two tests but it's late and I might overlook something ;)

I would keep it as is but use a GenericRAII instead, since it is only used where it is explicitly set before.